Code/GraphMol/SmilesParse/SmilesParse.cpp

```diff
@@ -368,7 +368,7 @@ RWMol *SmilesToMol(const std::string &smiles,
     std::string::const_iterator pos = cxPart.cbegin();
     try {
       SmilesParseOps::parseCXExtensions(*res, cxPart, pos);
-    } catch (const SmilesParseException &) {
+    } catch (...) {
```

Comment:
What else is being thrown?  I admit to feeling a bit nervous ignoring any exception if we don't have strict smiles set.

Reply:
we don't ignore the exception. We just catch it so that we can call `delete res` before throwing it again if we are doing strict parsing